app/components/notifications/notifications.js

```diff
@@ -0,0 +1,29 @@
+import { Controller } from 'stimulus';
+
+export default class extends Controller {
+  connect() {
+    // When Turbo renders a frame it will replace only contents that are
+    // within that particular frame. In most cases, flash messages/notifications
+    // are rendered outside of frames. To work around this, we hook into the
+    // Turbo lifecycle and manually replace the current notifications container
+    // with the one from the new response.
+    this.frameRenderHandler = event =>
+      this.replaceNotifications(event.detail.fetchResponse);
+    window.addEventListener('turbo:frame-render', this.frameRenderHandler);
```

Comment:
might be possible to use a Stimulus [global event action](https://stimulus.hotwired.dev/reference/actions#global-events) for this

Reply:
Implemented in b7db1e7.